scripts/source-maintenance.sh

```diff
@@ -84,27 +101,37 @@ done
 # an error code seen by a KeepGoing-aware command (or zero)
 SeenErrors=0
 
-
 if ! git diff --quiet; then
 	echo "There are unstaged changes. This script may modify sources."
 	echo "Stage changes to avoid permanent losses when things go bad."
 	exit 1
 fi
 
-# On squid-cache.org we have to use the python scripted md5sum
-HOST=`hostname`
-if test "$HOST" = "squid-cache.org" ; then
-	MD5="md5"
+for Checksum in md5sum md5 shasum sha1sum
+do
+    if "$Checksum" </dev/null >/dev/null 2>/dev/null ; then
+        break
+    fi
+done
+echo "detected checksum program $Checksum"
+
+if [ "x$ASTYLE" != "x" ] ; then
+    if ${ASTYLE} --version >/dev/null 2>/dev/null ; then
+        :
+    else
+        echo "ERROR: cannot run user-supplied astyle ${ASTYLE}"
+    fi
 else
-	MD5="md5sum"
+    for AttemptedBinary in astyle-${TargetAstyleVersion} astyle
+    do
+        if $AttemptedBinary --version >/dev/null 2>/dev/null ; then
+            ASTYLE=$AttemptedBinary
+            echo "detected astyle program ${ASTYLE}"
+            break
+        fi
+    done
 fi
 
-${ASTYLE} --version >/dev/null 2>/dev/null
-result=$?
-if test $result -gt 0 ; then
-	echo "ERROR: cannot run ${ASTYLE}"
-	exit 1
-fi
 ASVER=`${ASTYLE} --version 2>&1 | grep -o -E "[0-9.]+"`
```

Comment:
AFAICT, after PR changes, `${ASTYLE}` may be empty here. We will be running a command called `--version` in that case, which will be confusing. I suspect the PR is missing an ASTYLE assignment (to the default "astyle" value) earlier, when/if/in-case the AttemptedBinary test fails.

Reply:
Good catch.
Reworking to ${ASTYLE:-false}; this will trigger the "cannot run astyle" branch and fail